docs/articles/documentation/using-testcafe/common-concepts/screenshots-and-videos.md

```diff
@@ -0,0 +1,308 @@
+
+### Enable Video Recording
+
+Use either of the following to enable video recording:
+
+* the [--video](../command-line-interface.md#--video-basepath) command line flag,
+
+    ```sh
+    testcafe chrome test.js --video artifacts/videos
+    ```
+
+* the [runner.video](../programming-interface/runner.md#video) API method,
+
+    ```js
+    runner.video('artifacts/videos');
+    ```
+
+* the [videoPath](../configuration-file.md#videopath) configuration file property.
+
+    ```json
+    {
+        "videoPath": "artifacts/videos"
+    }
+    ```
+
+You must provide the base path where TestCafe stores videos to this flag, method or property. See [Where Screenshots and Videos Are Saved](#where-screenshots-and-videos-are-saved) for more information.
+
+TestCafe records all the tests and saves the recording of each test in a separate file. To change this behavior, use the `failedOnly` and `singleFile` [video options](#basic-video-options).
+
+### Basic Video Options
+
+TestCafe supports the following video options:
+
+Option | Type | Description | Default Value
+------ | ---- | ----------- | --------------
+`failedOnly` | Boolean | `true` to record only failed tests; `false` to record all tests. | `false`
+`singleFile` | Boolean | `true` to save the entire recording as a single file; `false` to create a separate file for each test. | `false`
+`ffmpegPath` | String | The path to the FFmpeg codec executable. | Auto-detected
+`pathPattern` | String | A pattern that defines how TestCafe composes the relative path to a video file and the file name. See [Where Screenshots and Videos Are Saved](#where-screenshots-and-videos-are-saved). | See [Default Path Patterns](#default-path-patterns).
+
+You can specify video options in either of the following ways:
+
+* the [--video-options](../command-line-interface.md#--video-options-optionvalueoption2value2) command line flag,
+
+    ```sh
+    testcafe chrome test.js --video artifacts/videos --video-options singleFile=true,failedOnly=true,pathPattern=${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4
+    ```
+
+* the `options` parameter of the [runner.video](../programming-interface/runner.md#video) API method,
+
+    ```js
+    runner.video('artifacts/videos', {
+        singleFile: true,
+        failedOnly: true,
+        pathPattern: '${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4'
+    });
+    ```
+
+* the [videoOptions](../configuration-file.md#videooptions) configuration file property.
+
+    ```json
+    {
+        "videoOptions": {
+            "singleFile": true,
+            "failedOnly": true,
+            "pathPattern": "${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4"
+        }
+    }
+    ```
+
+### Video Encoding Options
+
+You can specify all the options supported by the FFmpeg library. Refer to [the FFmpeg documentation](https://ffmpeg.org/ffmpeg.html#Options) for information about the available options.
+
+To provide video encoding options, use either of the following:
```

Comment:
either of the following what?

Reply:
So, we can say 'do one of the following:' - you usually approve it - but 'use either of the following:' requires a noun? In this case, I think 'options' is the right one.

Reply:
To provide video encoding options, use either of the following options: